src/XR/features/WebXRFeaturePointSystem.ts

```diff
@@ -0,0 +1,155 @@
+import { WebXRFeaturesManager, WebXRFeatureName } from "../webXRFeaturesManager";
+import { WebXRSessionManager } from "../webXRSessionManager";
+import { Observable } from "../../Misc/observable";
+import { Vector3 } from "../../Maths/math.vector";
+import { WebXRAbstractFeature } from "./WebXRAbstractFeature";
+
+/**
+ * A babylon interface for a "WebXR" feature point.
+ * Represents the position and confidence value of a given feature point.
+ */
+export interface IWebXRFeaturePoint {
+    /**
+     * Represents the position of the feature point in world space.
+     */
+    position : Vector3;
+    /**
+     * Represents the confidence value of the feature point in world space. 0 being least confident, and 1 being most confident.
+     */
+    confidenceValue : number;
+    /**
+     * The ID of the feature point, stable across frames.
+     */
+    id : number;
+}
+
+/**
+ * The feature point system is used to detect feature points from real world geometry.
+ * This feature is currently experimental and only supported on BabylonNative, and should not be used in the browser.
+ * The newly introduced API can be seen in webxr.nativeextensions.d.ts and described in FeaturePoints.md.
+ */
+export class WebXRFeaturePointSystem extends WebXRAbstractFeature {
+    private _enabled: boolean = false;
+    /**
+     * The module's name
+     */
+    public static readonly Name = WebXRFeatureName.FEATURE_POINTS;
+    /**
+     * The (Babylon) version of this module.
+     * This is an integer representing the implementation version.
+     * This number does not correspond to the WebXR specs version
+     */
+    public static readonly Version = 1;
+    /**
+     * Observers registered here will be executed whenever new feature points are available (on XRFrame while the session is tracking).
+     * Will notify the observers about which feature points have been updated.
+     */
+    public onFeaturePointsUpdatedObservable: Observable<number[]> = new Observable();
+    /**
+     * The currrent feature point cloud maintained across frames.
+     */
+    private _featurePointCloud: Array<IWebXRFeaturePoint> = [];
+    public readonly featurePointCloud: Array<IWebXRFeaturePoint> = this._featurePointCloud;

```

Comment:
Should this be a dictionary of id -> feature point instead?  Right now this is dependent on the IDs being strictly ordered.  Since we handle this on the BabylonNative side this is okay, but maybe it would be safer not to make this assumption.

Reply:
JS noob here but from an API design perspective, we should make it safe and intuitive to use. Because otherwise, I would expect the expected usage here is that I have to iterate through array to find the ID of feature point that actually got added/updated making this a On2 operation.

Making a dictionary of <id, feature point> would make the use case of simply iteration more cumbersome though. Are there any practical implications in JS here?

Reply:
I think from a practical implementation it's about the same.  You're correct iterating through feature points themselves would be more cumbersome, and I think there's a little bit of an efficiency overhead we save by using arrays, but I'm not too sure myself.  I'll leave this question open for now for more opinions.